Source/WTF/wtf/Vector.h

```diff
@@ -108,217 +94,113 @@ struct VectorInitializer<true, false, T>
     template<typename... Args>
     static void initializeWithArgs(T* begin, T* end, Args&&... args)
     {
+        static_assert(VectorTraits<T>::needsInitialization);
+        static_assert(!VectorTraits<T>::canInitializeWithMemset);
+
         for (T *cur = begin; cur != end; ++cur)
             new (NotNull, cur) T(args...);
```

Comment:
I think we can use `std::uninitialized_value_construct` or `std::ranges_uninitialized_value_construct` here instead of writing our own loop, although we will need to verify that it's still equally efficient.